week1/homework/test/server.test.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+const request = require('request-promise-native');
+
+const {
+  StatusCodeError
+} = require('request-promise-core/errors');
+
+const {
+  createServer
+} = require('../src/server');
+
+const TIMEOUT = 100;
+
+let port = 60000;
+
+function url(_port, method) {
+  return `http://localhost:${_port}/${method}`;
+}
+
+function test(_port, state, ...methods) {
+  it(`/${methods.join(', /')} returns ${state}`, async() => {
+    expect.assertions(1);
+
+    const server = await createServer(_port);
+    let response;
+    for (const method of methods)
+      try {
+        response = await request({
+          url:     url(_port, method),
+          timeout: TIMEOUT
+        });
+      }
+      catch (err) {
+        server.close();
+        throw err;
+      }
+    server.close();
+
+    const parsed = JSON.parse(response);
+
+    expect(parsed).toEqual({
+      state
+    });
```

Comment:
Do you also want to test for `Content-Type application/json` in the response header?

Reply:
Good idea.